### ISSUE TITLE
psalm-assert-if-true with boolean properties

### ISSUE THREAD

**Issue by franmomu** [vimeo/psalm#7891]
It would be nice if to be able to apply `psalm-assert-if-true|false` in boolean properties, something like:

https://psalm.dev/r/0d9e50fd84

**Comment by AndrolGenhald** [comment]
I don't think this is really the intent of assertions. A function asserting something should check that that thing is true. For this to be safe Psalm would need to always know the value of `A::$isVersioned` for every instance, which is more like a template, but unfortunately templates also don't work very well for this use-case.

I've run into scenarios where something like this would be useful as well, but I don't think this is the solution.

**Comment by rzvc** [comment]
I also want this, and not only for properties, but for any bool. From the user code's POV, there's no difference between `if (foo())` and `if ($foo)`.

The most common use case for this is when one bool indicates that a range of other stuff is valid:
```php
/**
 * @return null | array{ foo: Foo, bar: Bar }
 */
function test() : mixed
{
	/**
	 * @psalm-assert Foo $foo
	 * @psalm-assert Bar $bar
	 */
	$b = false;

	if (...)
	{
		$foo = new Foo();
		$bar = new Bar();

		$b = true;
	}

	return $b ? ['foo' => $foo, 'bar' => $bar] : null;
}
```

**Comment by still-dreaming-1** [comment]
This request is within the spirit of already existing features. It is similar to the example in https://psalm.dev/docs/annotating_code/adding_assertions/ where `->hasException()` is called to know that `->getException()` will return `Exception` and not `?Exception`. However, there is a note under that example which says 

> Please note that the example above only works if you enable method call memoization in the config file or annotate the class as immutable.

Therefore I think this feature might only make sense in a similar scenario, such as when the class is immutable or the property is readonly. Here is an example that illustrates how this feature could be used/useful: https://psalm.dev/r/8aecb78824
